slatedb/src/manifest/store.rs

```diff
@@ -621,33 +937,17 @@ impl ManifestStore {
         &self,
         id_range: R,
     ) -> Result<Vec<ManifestFileMetadata>, SlateDBError> {
-        let manifest_path = &Path::from("/");
-        let mut files_stream = self.object_store.list(Some(manifest_path));
-        let mut manifests = Vec::new();
-
-        while let Some(file) = match files_stream.next().await.transpose() {
-            Ok(file) => file,
-            Err(e) => return Err(SlateDBError::from(e)),
-        } {
-            match self.parse_id(&file.location, "manifest") {
-                Ok(id) if id_range.contains(&id) => {
-                    manifests.push(ManifestFileMetadata {
-                        id,
-                        location: file.location,
-                        last_modified: file.last_modified,
-                        size: file.size as u32,
-                    });
-                }
-                Err(_) => warn!(
-                    "unknown file in manifest directory [location={:?}]",
-                    file.location
-                ),
-                _ => {}
-            }
+        let files = self.inner.list(id_range).await?;
+        let mut out = Vec::with_capacity(files.len());
```

Comment:
I think it might be more idiomatic to do self.inner.list(id_range).map(|| ManifestFileMetadata(..)).collect::Vec<>().